launcher/common/vllm/query.sh

```
@@ -0,0 +1,129 @@
+if [ -z ${SLURM_ARRAY_TASK_COUNT} ]; then
+    TASK_COUNT=1
+else
+    echo "SLURM_ARRAY_TASK_COUNT ${SLURM_ARRAY_TASK_COUNT}"
+    TASK_COUNT=${SLURM_ARRAY_TASK_COUNT}
+fi
+
+# Parse --model and split remaining args on "--".
+# --model is consumed here; args before "--" go to vllm serve, args after go to query.py.
+MODEL=""
+SERVE_EXTRA_ARGS=()
+QUERY_ARGS=(--shard-id-begin ${TASK_ID} --shard-id-step ${TASK_COUNT})
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)